spec/GraphQLOverHTTP.md

```diff
@@ -112,72 +169,133 @@ http://example.com/product/graphql
+
+- {query} - A Document containing GraphQL Operations and Fragments to execute.
+- {operationName} - (_Optional_): The name of the Operation in the Document to
+  execute.
+- {variables} - (_Optional_): Values for any Variables defined by the Operation.
+- {extensions} - (_Optional_): This entry is reserved for implementors to extend
+  the protocol however they see fit.
 
-* {query} - A Document containing GraphQL Operations and Fragments to execute.
-* {operationName} - (*Optional*): The name of the Operation in the Document to execute.
-* {variables} - (*Optional*): Values for any Variables defined by the Operation.
-* {extensions} - (*Optional*): This entry is reserved for implementors to extend the protocol however they see fit.
+Note: When comparing against
+[request in the GraphQL specification](https://spec.graphql.org/draft/#request)
+you should note the schema and "initial value" are not included in the
+GraphQL-over-HTTP request, they are handled by the server based on the URL used.
 
-Note: Be aware that `query` is a misleading name as it can contain a string describing multiple operations, each of which may be a query, mutation or subscription. A better name would have been `document`, but the term `query` is well established.
+Note: Be aware that `query` is a misleading parameter name as its value is a
+string describing one or more operations, each of which may be a query, mutation
+or subscription. A better name would have been `document`, but the term `query`
+is well established.
 
-Note: depending on the serialization format used, values of the aforementioned parameters can be
-encoded differently but their names and semantics must stay the same.
+Note: Depending on the serialization format used, values of the aforementioned
+parameters can be encoded differently but their names and semantics must stay
+the same.
 
-Note: specifying `null` in JSON (or equivalent values in other formats) as values for optional request parameters is equivalent to not specifying them at all.
+Note: Specifying `null` in JSON (or equivalent values in other formats) as
+values for optional request parameters is equivalent to not specifying them at
+all.
 
-Note: {variables} and {extensions}, if set, must have a map as its value.
+Note: {variables} and {extensions}, if set, must have a map as their value.
+
+### JSON Encoding
+
+When encoded in JSON, a GraphQL-over-HTTP request is a JSON object (map), with
+the properties specified by the GraphQL-over-HTTP request:
+
+- {query} - the string representation the Source Text of the Document as
+  specified in
+  [the Language section of the GraphQL specification](https://spec.graphql.org/draft/#sec-Language).
+- {operationName} - an optional string
+- {variables} - an optional object (map), the keys of which are the variable
+  names and the values of which are the variable values
+- {extensions} - an optional object (map)
+
+## Accept
```

Comment:
This spec bounces around a lot.  If I were implementing a client or a server separately, it might be hard to determine what it means to be compliant.  Creating a separation might make it easier.

Reply:
Ah, great comment. Believe it or not I was trying not to change too much so didn't change the section titles :laughing: But yeah, I think I was thinking of it in terms of client/server rather than request/response a lot of the time, it feels more natural. I'll think about it...